python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -97,7 +103,25 @@ def op_config(self) -> Any:
         """The parsed config specific to this op."""
 
 
-class OpExecutionContext(AbstractComputeExecutionContext):
+class OpExecutionContextMetaClass(AbstractComputeMetaclass):
+    def __instancecheck__(cls, instance) -> bool:
+        # This makes isinstance(context, OpExecutionContext) throw a deprecation warning when
+        # context is an AssetExecutionContext. This metaclass can be deleted once AssetExecutionContext
+        # has been split into it's own class in 1.7.0
+        if type(instance) is AssetExecutionContext:
```

Comment:
im confused by this comment and line 111, seems like it will raise only if `isinstance` target is `AssetExecutionContext` ? 

can we add some tests to demonstrate when it does / doesn't warn?

Reply:
see https://github.com/dagster-io/dagster/pull/16762 for tests. i'm working on a way to scoop the test into this PR, but it currently needs the changes in the stacked PR

I think the nuance here is that when you do `isinstance(context, OpExecutionContext`) the `instance` in this method is `context`

Reply:
ok got the test pulled into this PR. it's a bit uglier, but i prefer only having one PR to revert if we need to

Reply:
this gets inherited by `AssetExecutionContext` as well right? So in that case I think well need to be checking `cls` too to prevent warning on [1]